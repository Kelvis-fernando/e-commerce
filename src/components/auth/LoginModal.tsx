import { AuthModalContainer } from "../../styles/auth/loginModal";
import TitleAndSubtitle from "../layout/TitleAndSubtitle";
import Modal from "../layout/Modal";
import RegisterModal from "./RegisterModal";
import ForgotPasswordModal from "./ForgotPasswordModal";
import useLoginModal from "../../hooks/auth/useLoginModal";
import useAuth from "../../hooks/auth/useAuth";
import Toast from "../Toast";
import useToast from "../../hooks/useToast";
import { useState } from "react";

const LoginModal = () => {
  const {
    registerModalOPen,
    setRegisterModalOPen,
    forgotPasswordModalOPen,
    setForgotPasswordModalOPen,
  } = useLoginModal();
  const { showToast, handleButtonClick, handleCloseToast } = useToast();
  const [errorToast, setErrorToast] = useState(false);
  const { handleGetAuthFormData } = useAuth(handleButtonClick, setErrorToast);

  return (
    <>
      <AuthModalContainer>
        <TitleAndSubtitle
          title="OlÃ¡, seja bem vindo"
          subtitle="Entre na sua conta"
        />
        <form onSubmit={(event) => handleGetAuthFormData(event, "login")}>
          <label htmlFor="email">E-mail</label>
          <input type="email" name="email" id="email" />
          <label htmlFor="password">Senha</label>
          <input type="password" name="password" id="password" />
          <button type="submit">Entrar</button>
        </form>
        <span>
          <h2 onClick={() => setRegisterModalOPen(true)}>Cadastre-se</h2>
          <h2 onClick={() => setForgotPasswordModalOPen(true)}>
            Esqueceu a senha?
          </h2>
        </span>
      </AuthModalContainer>
      <Modal
        isOpen={registerModalOPen}
        onClose={() => setRegisterModalOPen(false)}
      >
        <RegisterModal onBack={setRegisterModalOPen} />
      </Modal>
      <Modal
        isOpen={forgotPasswordModalOPen}
        onClose={() => {
          setForgotPasswordModalOPen(false);
        }}
      >
        <ForgotPasswordModal onBack={setForgotPasswordModalOPen} />
      </Modal>
      {showToast && (
        <Toast
          message={"Login feito com sucesso!"}
          onClose={handleCloseToast}
        />
      )}
      {errorToast && (
        <Toast
          message={"Erro com os dados do usuario"}
          typeOfToast="error"
          onClose={() => setErrorToast(false)}
        />
      )}
    </>
  );
};

export default LoginModal;
