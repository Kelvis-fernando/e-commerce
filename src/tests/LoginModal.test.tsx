import React from "react";
import { render } from "@testing-library/react";
import LoginModal from "../components/auth/LoginModal";

describe("LoginModal", () => {
  test("should render and match snapshot", () => {
    const { container } = render(<LoginModal />);

    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
