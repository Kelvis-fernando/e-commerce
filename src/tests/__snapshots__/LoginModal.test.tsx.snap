// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginModal should render and match snapshot 1`] = `
<div>
  <div
    class="sc-bcXHqe hlQNyq"
  >
    <div
      class="sc-gswNZR dKCtgi"
    >
      <h1>
        Ol√°, seja bem vindo
      </h1>
      <h3>
        Entre na sua conta
      </h3>
    </div>
    <form>
      <label
        for="email"
      >
        E-mail
      </label>
      <input
        id="email"
        name="email"
        type="email"
      />
      <label
        for="password"
      >
        Senha
      </label>
      <input
        id="password"
        name="password"
        type="password"
      />
      <button
        type="submit"
      >
        Entrar
      </button>
    </form>
    <span>
      <h2>
        Cadastre-se
      </h2>
      <h2>
        Esqueceu a senha?
      </h2>
    </span>
  </div>
</div>
`;
